pliers/extractors/audio.py

```diff
@@ -488,4 +495,126 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
     For details on argument specification visit:
     https://librosa.github.io/librosa/effect.html.'''
 
+
     _feature = 'percussive'
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+    
+    ''' Extract probability of 521 audio event classes based on AudioSet
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the audio segment (in seconds) on which label 
+        extraction is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If None, all labels (or all in labels) are returned.
+        Top_n and labels are mutually exclusive arguments.
+    labels (list): specifies subset of labels for which probabilities 
+        are to be returned. If None, all labels (or top_n) are returned.
+        The full list of labels is available in the audioset/yamnet 
+        repository (see yamnet_class_map.csv).
+    weights_path (optional): full path to model weights file. If not provided,
+        weights from pretrained YAMNet module are used.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'labels', 'weights_path',
+                       'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, labels=None,
+                 weights_path=None, yamnet_path=None, **yamnet_kwargs):
+        if yamnet_path is None:
```

Comment:
Is there a reason to allow the user to specify a custom yamnet path? I'm assuming this would be an alternate way to import yamnet... the risk there is that users probably won't want to use this unless they have reason to think something has changed in the yamnet implementation. And if something *has* changed in the yamnet implementation, there's a good chance it will imply an API change, which could result in things breaking for us. So I guess my inclination would be to remove this check and always point to our own installer.

Reply:
Agreed, unless we think sometimes our install would fail. But I think that should be considered a bug to fix, so no need to build a workaround yet.

Reply:
removed the option to specify custom `yamnet_path` (that was only to avoid downloading extra stuff in the user already has the yamnet or the whole tf repo somewhere else).

Reply:
Probably what we should ultimately do is settle on a unified way of handling downloaded models/files, so that users can do something like `pliers install yamnet --dir=/path/to/put/files` from the command line, where the storage location is optional (and would default to what we're currently using). Might be worth opening an issue. But since this is the only model like this right now (all the others have their own packages that handle setup themselves), let's not get sidetracked with that right now. This seems fine.